pkg/controller/masteruserrecord/masteruserrecord_controller.go

```diff
@@ -95,6 +101,111 @@ func (r *ReconcileMasterUserRecord) Reconcile(request reconcile.Request) (reconc
 		return reconcile.Result{}, err
 	}
 
-	// Do nothing for now
+	for _, account := range userRecord.Spec.UserAccounts {
+		err = r.ensureUserAccount(reqLogger, account, userRecord)
+		if err != nil {
+			reqLogger.Error(err, "unable to synchronize with member UserAccount")
+			return reconcile.Result{}, err
+		}
+	}
+
 	return reconcile.Result{}, nil
 }
+
+func (r *ReconcileMasterUserRecord) ensureUserAccount(log logr.Logger, recAccount toolchainv1alpha1.UserAccountEmbedded, record *toolchainv1alpha1.MasterUserRecord) error {
+	// get & check fed cluster
+	fedCluster, err := r.getMemberCluster(log, recAccount, record)
+	if err != nil {
+		return r.wrapErrorWithStatusUpdate(log, record, r.setStatusFailed(targetClusterNotReady), err,
+			"failed to get the member cluster '%s'", recAccount.TargetCluster)
+	}
+
+	// get UserAccount from member
+	nsdName := namespacedName(fedCluster.OperatorNamespace, record.Name)
+	userAccount := &toolchainv1alpha1.UserAccount{}
+	err = fedCluster.Client.Get(context.TODO(), nsdName, userAccount)
```

Comment:
I like the `fedCluster.Client.Get()`, it feels similar to getting object from normal client.